common/djangoapps/student/tests/test_events.py

```diff
@@ -0,0 +1,112 @@
+# -*- coding: utf-8 -*-
+"""
+Test that various events are fired for models in the student app.
+"""
+from django.test import TestCase
+
+from student.models import USER_SETTINGS_CHANGED_EVENT_NAME
+from student.tests.factories import UserFactory
+from util.testing import EventTestMixin
+
+
+class UserSettingsEventTestMixin(EventTestMixin):
+    """
+    Mixin for verifying that user setting events were emitted during a test.
+    """
+    def setUp(self):
+        super(UserSettingsEventTestMixin, self).setUp('util.model_utils.tracker')
+
+    def assert_user_setting_event_emitted(self, **kwargs):
+        """
+        Helper method to assert that we emit the expected user settings events.
+
+        Expected settings are passed in via `kwargs`.
+        """
+        self.assert_event_emitted(
+            USER_SETTINGS_CHANGED_EVENT_NAME,
+            table=self.table,  # pylint: disable=no-member
+            user_id=self.user.id,
+            **kwargs
+        )
+
+
+class TestUserProfileEvents(UserSettingsEventTestMixin, TestCase):
+    """
+    Test that we emit field change events when UserProfile models are changed.
+    """
+    def setUp(self):
+        super(TestUserProfileEvents, self).setUp()
+        self.table = 'auth_userprofile'
+        self.user = UserFactory.create()
+        self.profile = self.user.profile
+        self.reset_tracker()
+
+    def test_change_one_field(self):
+        """
+        Verify that we emit an event when a single field changes on the user
+        profile.
+        """
+        self.profile.year_of_birth = 1900
+        self.profile.save()
+        self.assert_user_setting_event_emitted(setting='year_of_birth', old=None, new=self.profile.year_of_birth)
+
+    def test_change_many_fields(self):
```

Comment:
Must: Include a test case for country in this class, since that field requires special handling.

Haven't looked yet to see if the test exists elsewhere, but I think this is the logical place.